doc/source/coding_style.rst

```diff
@@ -135,3 +135,173 @@ update any projects dependent on the API while still being treated as
 "first-class" users.  Note that due to the complexity of maintaining
 multiple "release branches" in a repository, the number of active
 release branches should be between one and three.
+
+Docstring Examples Best-Practice
+---------------------------------
+Defining docstring examples for methods and classes are extremely 
+useful. The examples give users an easy place to start when trying 
+out the API, by showing them exactly how to operate on a method or 
+class. By using ``doctest`` through ``pytest``, docstring examples can 
+also be used to perform regression testing to verify that the code is 
+executing as expected.
+
+Setting Up ``doctest``
+~~~~~~~~~~~~~~~~~~~~~~
+First, install ``pytest``:
+
+.. code::
+
+    pip install pytest
+
+Then, ``doctest`` can be run on any Python file by running:
+
+.. code::
+
+    pytest --doctest-modules file.py
+
+Here, ``doctest`` will search for any examples in the docstrings and will then
+execute them to verify that they function as written.
+
+Using ``pytest`` Fixtures
+~~~~~~~~~~~~~~~~~~~~~~~~~
+To define a set up sequence before the ``doctest`` run or before a given 
```

Comment:
```suggestion
To define a setup sequence before the ``doctest`` run or before a given 
```
I think "setup" makes more sense here as a noun vs "set up" which is two words as a verb.

Reply:
Yes, you're right. I just keep writing setup since I am so used to the AEDT concept "Analysis Setup"!